docs/declaration_files.md

```diff
@@ -0,0 +1,86 @@
+# Type definitions for third party libraries
+
+You can create declaration files to annotate the types of third-party Lua
+modules, including C Lua modules. To do so, create a file with the .d.tl
+extension and require it as normal, i.e. `local lfs = require("lfs")`.
+
+Types defined in this module will will be used as a source of type information 
+checking with `tl check`, even though the real Lua module will be loaded
+instead when requiring the module from Lua or `tl run`.
+
+## Visibility
+
+There are two ways to define these types:
+
+### Composite Types
+
+```
+local record MyCompositeType
+   record MyPointType
+      x: number
+      y: number
+   end
+
+   center: MyPointType
+    -- insert more stuff here
+end
+
+return MyCompositeType
+```
+
+This will mean that references to `MyPointType` must be qualified (or locally declared) as
+`MyCompositeType.MyPointType`.
+
+### Global Types
+
+```
+global record MyPointType
+   x: number
+   y: number
+end
+
+global record MyCompositeType
+   center: MyPointType
+end
+```
+
+These can now be used unqualified in any file that requires them.
+
+#### Global environment definition
+
+If the third party library is "globally available" in your execution environment,
+i.e., if you do not need to explicitly `require` it in your code, you can tell the compiler
+to predefine it into its own type checking environment.
+
+For instance, assuming you have the following declaration file, called `love-example.d.tl`:
+
+```
+-- love-example.d.tl
+
+global record love
+   record graphics
+      print: function(text: string, x: number, y: number)
+   end
+end
+```
+
+You can predefine the `love` module by creating a `tlconfig.lua` file at the root of your project:
+
+```
+-- tlconfig.lua
+
+return {
+   global_env_def = "love-example"
+}
+```
+
+This makes the compiler aware of the `love` global record.
+You may then freely refer to this variable in your code:
+
+```
+-- main.tl
+
+love.graphics.print("hello!", 0, 0)
+```
+
+You can find more information about global environment definition in the [compiler options](compiler_options.md#global-environment-definition) page.
```

Comment:
Maybe it would make sense to get rid of the 'Global environment definition' section in **compiler_options.md**, and keep all the information related to declaration files in **declaration_files.md**.

Reply:
Maybe turn it into a short sentence with a link to this page?

Another suggestion: this page could benefit from a link to teal-types, so that people who land here first don't end up rewriting some .d.tl which already exists!

Reply:
Done!

I have also rephrased the text in `declaration_files.md` to make it look more like the old 'Global environment definition' section  in `compiler_options.md`.

Feel free to merge if it reads fine :)

Reply:
Great, looks good! Merged!!